core/include/detray/propagator/track.hpp

```diff
@@ -28,10 +28,16 @@ struct bound_track_parameters {
         : _surface_link(sf_idx), _vector(params), _covariance(cov) {}
 
     DETRAY_HOST_DEVICE
-    vector_type& vector() { return _vector; }
+    vector_type vector() const { return _vector; }
```

Comment:
Will you use and modify the vector on return, or why won't you give it back as `const vector_type&` ?

Reply:
Actually not. You are right. const reference should be better choice.
I change the output types `vector()` and `covariance()` to const reference.